services/src/test/java/org/keycloak/vault/PlainTextVaultProviderTest.java

```diff
@@ -0,0 +1,80 @@
+package org.keycloak.vault;
+
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for {@link PlainTextVaultProvider}.
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProviderTest {
+
+    @Test
+    public void shouldObtainSecret() throws Exception {
+        //given
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(Scenario.EXISTING.getPath());
+
+        //when
+        VaultRawSecret secret1 = provider.obtainSecret("key1");
+
+        //then
+        assertNotNull(secret1);
+        assertNotNull(secret1.getRawSecret().get());
+        assertEquals("secret1", new String(secret1.getRawSecret().get().array()));
+    }
+
+    @Test
+    public void shouldReturnEmptyOptionalOnMissingSecret() throws Exception {
+        //given
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(Scenario.EXISTING.getPath());
+
+        //when
+        VaultRawSecret secret = provider.obtainSecret("non-existing-key");
+
+        //then
+        assertNotNull(secret);
+        assertFalse(secret.getRawSecret().isPresent());
+    }
+
+    @Test
+    public void shouldOperateOnNonExistingVaultDirectory() throws Exception {
+        //given
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(Scenario.NON_EXISTING.getPath());
+
+        //when
+        VaultRawSecret secret = provider.obtainSecret("non-existing-key");
+
+        //then
+        assertNotNull(secret);
+        assertFalse(secret.getRawSecret().isPresent());
+    }
+
+    @Test
+    public void shouldReflectChangesInASecretFile() throws Exception {
+        //given
+        Path temporarySecretFile = Files.createTempFile("vault", null);
+        Path vaultDirectory = temporarySecretFile.getParent();
+        String secretName = temporarySecretFile.getFileName().toString();
+
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(vaultDirectory);
+
+        //when
+        Files.write(temporarySecretFile, "secret1".getBytes());
+        VaultRawSecret secret1 = provider.obtainSecret(secretName);
```

Comment:
This should use try-with-resources when obtaining the secret.
And actually it should try to read the same secret twice to make sure that in mmaped case, destroying does not mean overwriting the file

Reply:
That's a very good point. Let me also add a new test case for checking the destroying part.